### PR TITLE
add CS100 Spring 2024

### PR DESCRIPTION
**Pull request by TLNA114514**
add lecture slides; add hw; add hw sol; add exam; add quiz

### CODE REVIEW

**Comment by why-in-Shanghaitech** [comment]
Hi! Thank you very much for your contribution! I'll review this PR in a few days.

File: courses/CS100/meta.json [hunk 1]
```diff
@@ -2,6 +2,36 @@
...
+                "gongke",
+                "liuxiao",
+                "李钟楷",
+                "Qingcheng Zhao",
+                "张涵锐",
+                "Zhiyu Yang",
+                "张中瀚",
+                "Yunlong Feng",
+                "daihzh",
+                "张敏慎",
```
Comment (why-in-Shanghaitech) [inline courses/CS100/meta.json]: Use chinese character or space may lead to undefined behaviour. These ids are also missing from the people database.

**Review by why-in-Shanghaitech — CHANGES_REQUESTED** [review]
The only issue is about people id. I can help you with this, but I am afraid I'll be very busy in the following month (till 18th, August).

File: courses/CS100/meta.json (hunk 1 above)
Comment (why-in-Shanghaitech) [reply courses/CS100/meta.json]: image

Click here to create a new people in the database. The id is expected to be the email prefix (or something similar is okay). The name is what is displayed on the webpage.

**Comment by TLNA114514** [comment]
Alright, I've already finished it.However, I could not  know the exact name of the TA 'daihz'  , I decided to delete it in case that it may mislead others.

发自 Outlook<http://aka.ms/weboutlook>
________________________________
发件人: Haoyi Wu ***@***.***>
发送时间: 2024年7月21日 13:06
收件人: i-TechX/iTechX ***@***.***>
抄送: TLNA114514 ***@***.***>; Author ***@***.***>
主题: Re: [i-TechX/iTechX] add CS100 Spring 2024 (PR #121)


@why-in-Shanghaitech commented on this pull request.

________________________________

In courses/CS100/meta.json<https://github.com/i-TechX/iTechX/pull/121#discussion_r1685641377>:

> +                "李钟楷",
+                "Qingcheng Zhao",
+                "张涵锐",
+                "Zhiyu Yang",
+                "张中瀚",
+                "Yunlong Feng",
+                "daihzh",
+                "张敏慎",


image.png (view on web)<https://github.com/user-attachments/assets/a54d6d6c-e36a-40fd-b23b-ff3394ad29ce>

Click here to create a new people in the database. The id is expected to be the email prefix (or something similar is okay). The name is what is displayed on the webpage.

―

**Review by why-in-Shanghaitech — APPROVED** [review]
LGTM, need to manually resolve the conflict
There is still one Chinese person id. I will fix it before merging.

**Comment by why-in-Shanghaitech** [comment]
Again, thank you for your great contribution! These contents are now available at iTechX.

**Comment by TLNA114514** [comment]
Thank you very much！

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Haoyi Wu ***@***.***>
发送时间: Tuesday, August 20, 2024 10:44:01 AM
收件人: i-TechX/iTechX ***@***.***>
抄送: TLNA114514 ***@***.***>; Author ***@***.***>
主题: Re: [i-TechX/iTechX] add CS100 Spring 2024 (PR #121)


Again, thank you for your great contribution! These contents are now available at iTechX.

―

**Pull request merged**